test/e2e/nodeportlocal_test.go

```diff
@@ -0,0 +1,389 @@
+		t.Fatalf("Error when creating Pod '%s': %v", clientName, err)
+	}
+
+	_, err = nplTestData.podWaitForIPs(defaultTimeout, clientName, testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientName, err)
+	}
+
+	antreaPod, err := nplTestData.getAntreaPodOnNode(node)
+	r.Nil(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
+
+	checkForNPLRuleInIPTABLES(t, nplTestData, r, nplAnnotations, antreaPod, testPodIP.ipv4.String(), true)
+	validatePortInRange(t, nplAnnotations, defaultStartPort, defaultEndPort)
+	checkTrafficForNPL(t, nplTestData, r, nplAnnotations, clientName)
+
+	nplTestData.deletePod(testPodName)
+	checkForNPLRuleInIPTABLES(t, nplTestData, r, nplAnnotations, antreaPod, testPodIP.ipv4.String(), false)
+}
+
+// TestNPLMultiplePodsAndAgentRestart tests NodePortLocal functionalities after Antrea Agent restart.
+// - Create Multiple Nginx Pods.
+// - Restart Antrea Agent Pods.
+// - Verify Pod Annotation, IPTABLES rules and traffic to test Pod.
+func TestNPLMultiplePodsAgentRestart(t *testing.T) {
+	skipIfNotIPv4Cluster(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	enableNPLInConfigmap(t, data)
+	r := require.New(t)
+
+	annotation := make(map[string]string)
+	annotation[nplSvcAnnotation] = "true"
+	ipFamily := corev1.IPv4Protocol
+	data.createNginxClusterIPServiceWithAnnotation(false, &ipFamily, annotation)
+
+	node := nodeName(0)
+	var testPods []string
+
+	for i := 0; i < 4; i++ {
+		testPodName := randName("test-pod-")
+		testPods = append(testPods, testPodName)
+		err = data.createNginxPod(testPodName, node)
+		r.Nil(err, "Error Creating test Pod: %v", err)
+	}
+
+	clientName := randName("test-client-")
+	if err := data.createBusyboxPodOnNode(clientName, node); err != nil {
+		t.Fatalf("Error when creating Pod '%s': %v", clientName, err)
+	}
+	_, err = data.podWaitForIPs(defaultTimeout, clientName, testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientName, err)
+	}
+
+	if err := data.restartAntreaAgentPods(defaultTimeout); err != nil {
+		t.Fatalf("Error when restarting Antrea: %v", err)
+	}
+
+	antreaPod, err := data.getAntreaPodOnNode(node)
+	r.Nil(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
+
+	for _, testPodName := range testPods {
+		nplAnnotationString, testPodIP := getNPLAnnotation(t, data, r, testPodName)
+		var nplAnnotations []k8s.NPLAnnotation
+		json.Unmarshal([]byte(nplAnnotationString), &nplAnnotations)
+
+		checkForNPLRuleInIPTABLES(t, data, r, nplAnnotations, antreaPod, testPodIP.ipv4.String(), true)
+		validatePortInRange(t, nplAnnotations, defaultStartPort, defaultEndPort)
+		checkTrafficForNPL(t, data, r, nplAnnotations, clientName)
+
+		data.deletePod(testPodName)
+		checkForNPLRuleInIPTABLES(t, data, r, nplAnnotations, antreaPod, testPodIP.ipv4.String(), false)
```

Comment:
maybe this is not needed, we test this in earlier tests?

Reply:
Removed.